docs/jailer.md

```diff
@@ -277,6 +277,13 @@ Note: default value for `<api-sock>` is `/run/firecracker.socket`.
   associated with `--daemonize` runs towards the end, instead of the very
   beginning. We are working on adding better logging capabilities.
 
+### Notes
+
+- The recommended method to get Firecracker process's pid is to use
+  `--new-pid-ns` flag and read Firecracker's pid from the `firecracker.pid` file
+  present in the jailer's root directory. This is also the recommended method to
+  fetch Firecracker pid if the pid is needed to kill Firecracker process.
+
```

Comment:
Mhh, I'm not entirely comfortable with saying that `--new-pid-ns` is the recommended way to get the pid, because that implies that the functionality of `--new-pid-ns` is "write a pid file" (but its not: it spawns firecracker into a new pid namespace, which might not be desired by the customer). So I would instead put it as a known issue (both here and on the 1.6 branch) to say something like "When passing the `--daemonize` option to Firecracker without the `--new-ns-pid` option, the Firecracker process will have a different PID than the jailer process, yet fail to write a pid file. To work around this, please see if using the `--new-ns-pid` flag is an option". 

And then when we fix the issue on main by having Firecracker actually write the PID file, remove the note again (or just fix the issue straightaway and never add a note to main, only to 1.6 xP)

Reply:
> because that implies that the functionality of --new-pid-ns is "write a pid file"
I don't think it does because we explain what `--new-pid-ns` does in usage and observation section twice. If someone sees this note they will check the other 2 to understand what it is.

> So I would instead put it as a known issue (both here and on the 1.6 branch)
That was the intention but I named the section incorrectly. Yes, the plan is to push on 1.6 as well once this is done.

> or just fix the issue straightaway and never add a note to main, only to 1.6 xP
I didn't want to rush into doing this because this is not a blocked and we have other priority work. But we'll pick this up soon next week :) .

I'll rename the section to known limitation and add the line to mention the change in pids, thank !

Reply:
done, more details https://github.com/firecracker-microvm/firecracker/pull/4436#discussion_r1484150881 
Thanks!

Reply:
thanks! :)